R/footer.R

```diff
@@ -200,3 +229,53 @@ footer <- function(full = FALSE, links = NULL) {
   )
   attachDependency(govFooter)
 }
+
+#' Create a footer link for use in `footer()` function
+#'
+#' @param link Character string containing either link text or url
+#' @param link_name Name of a link where a URL has been provided in link_text
+#'
+#' @returns HTML tag list item
+#'
+#' @keywords internal
+#' @examples
+#' # Internal (i.e. within dashboard) link
+#' shinyGovstyle:::footer_link("Cookie statement")
+#' # Named internal link
+#' shinyGovstyle:::footer_link("cookie_statement", "Cookies")
+#' # External link
+#' shinyGovstyle:::footer_link(
+#'   "https://www.gov.uk/government/organisations/government-digital-service",
+#'   "Government Digital Service"
+#' )
+footer_link <- function(
+    link,
+    link_name = NULL
+  ) {
+  if (grepl("^http|^www", link)) {
+    if (is.null(link_name)) {
+      warning("Link name provided is NULL for ", link)
+      link_name = link
+    }
+    shiny::tags$li(
+      class = "govuk-footer__inline-list-item",
+      external_link(
+        link,
+        link_name,
+        add_warning = FALSE
+      )
```

Comment:
Using external link overwrites the CSS classes (or at least adds more to it), which means it's blue rather than the expected black (e.g. - https://explore-education-statistics.service.gov.uk/)

![image](https://github.com/user-attachments/assets/f4d6d380-4053-402d-a8ca-2a494f546b32)